source/Octopus.Cli/Commands/CreateReleaseCommand.cs

```diff
@@ -1,199 +1,136 @@
+        }
+
+        public string ProjectName { get; set; }
+        public string ChannelName { get; set; }
+        public bool AutoChannel { get; set; }
+        public List<string> DeployToEnvironmentNames { get; set; }
+        public string VersionNumber { get; set; }
+        public string ReleaseNotes { get; set; }
+        public bool IgnoreIfAlreadyExists { get; set; }
+        public bool Force { get; set; }
+        public string VersionPreReleaseTag { get; set; }
+        public bool WhatIf { get; set; }
+
+        protected override void Execute()
+        {
+            if (string.IsNullOrWhiteSpace(ProjectName)) throw new CommandException("Please specify a project name using the parameter: --project=XYZ");
+
+            if (AutoChannel && !string.IsNullOrWhiteSpace(ChannelName)) throw new CommandException("Cannot specify --channel and --autochannel arguments");
+
+            if (WhatIf) Log.Info("What if: no release will be created.");
+
+            Log.Debug("Finding project: " + ProjectName);
+            var project = Repository.Projects.FindByName(ProjectName);
+            if (project == null)
+                throw new CouldNotFindException("a project named", ProjectName);
+
+            var plan = BuildReleasePlan(project);
 
+            string versionNumber;
+            if (!string.IsNullOrWhiteSpace(VersionNumber))
+            {
+                Log.Debug("Using version number provided on command-line.");
+                versionNumber = VersionNumber;
+            }
+            else if (!string.IsNullOrWhiteSpace(plan.ReleaseTemplate.NextVersionIncrement))
+            {
+                Log.Debug("Using version number from release template.");
+                versionNumber = plan.ReleaseTemplate.NextVersionIncrement;
+            }
+            else if (!string.IsNullOrWhiteSpace(plan.ReleaseTemplate.VersioningPackageStepName))
+            {
+                Log.Debug("Using version number from package step.");
+                versionNumber = plan.GetActionVersionNumber(plan.ReleaseTemplate.VersioningPackageStepName);
+            }
+            else
+            {
+                throw new CommandException("A version number was not specified and could not be automatically selected.");
+            }
+
+            if (plan.Steps.Any())
+            {
+                Log.Info("Release plan for release:    " + versionNumber);
+                Log.Info(plan.FormatAsTable());
+            }
+
+            if (plan.HasUnresolvedSteps())
+            {
+                throw new CommandException("Package versions could not be resolved for one or more of the package steps in this release. See the errors above for details. Either ensure the latest version of the package can be automatically resolved, or set the version to use specifically by using the --package argument.");
+            }
+
+            if (IgnoreIfAlreadyExists)

```

Comment:
I raised this in my first PR but to me it seems this logic is out of whack?  if `IgnoreIfAlreadyExists == true` then we check if a release exists and throw an exception if it does.  Shouldn't the if condition be `if (!IgnoreIfAlreadyExists)` ?